db.go

```diff
@@ -8,6 +8,8 @@ import (
 )
 
 const (
+	IMQSUserType AuthUserType = 0
```

Comment:
Place enum-like constants in their own "const" block. So AuthUserType constants go in their own block, and NullUserId in it's own block, etc. If it's just one constant, then it doesn't need a const block.

Find a different name for IMQSUserType, which doesn't use the word "IMQS". Until now, authaus hasn't had any explicit tie to IMQS, and it is good to keep it that way. So rather call it "BuiltinUser", or something like that.
Start the name with UserType, such as UserTypeLDAP, UserTypeBuiltin.

Since these constants are embedded in the database, you need to make that very clear. There must be comments where the constants (enums) are declared, and there needs to be an accompanying comment in the DB migration code explaining that the field takes values from the AuthUserType constants.

Do we really need to call it AuthUserType? Can it not just be UserType?